File: LeamiServices/Database/Entities/Review.cs
using Leami.Model.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Leami.Services.Database.Entities
{
    public class Review
    {
        public int ReviewId { get; set; }
        public int ReviewerUserId { get; set; }
        public short Rating { get; set; }
        public string? Comment { get; set; }
        public DateTime CreatedAt { get; set; }
        public User? ReviewerUser { get; set; } 
        public bool IsDeleted { get; set; } = false;

        public string? DeletionReason { get; set; } = string.Empty;


    }
}

File: Leami.UI/leami_desktop/lib/models/review.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:leami_desktop/models/user.dart';

part 'review.g.dart';

@JsonSerializable(explicitToJson: true)
class Review {
  int? reviewId;
  int? reviewerUserId;
  int? rating;
  String? comment;
  DateTime? createdAt;
  User? reviewerUser;
  bool? isDeleted;
  String? deletionReason;

  Review({
    this.reviewId,
    this.reviewerUserId,
    this.rating,
    this.comment,
    this.createdAt,
    this.reviewerUser,
    this.isDeleted,
    this.deletionReason,
  });

  factory Review.fromJson(Map<String, dynamic> json) => _$ReviewFromJson(json);
  Map<String, dynamic> toJson() => _$ReviewToJson(this);
}